code/modules/vehicles/mecha/combat/durand.dm

```diff
@@ -151,6 +151,7 @@ own integrity back to max. Shield is automatically dropped if we run out of powe
 	///To keep track of things during the animation
 	var/switching = FALSE
 	var/currentuser
+	resistance_flags = LAVA_PROOF | FIRE_PROOF //The shield should not take damage from fire or lava, that's the mech's job
```

Comment:
acid?

Reply:
I haven't been able to replicate the same power drain thing by throwing a bottle of acid at the shield. Also, isn't xeno spit acid?

Reply:
> acid?

Nevermind you're right, I got it to happen after using a grenade.